auth0/src/main/java/com/auth0/android/provider/OAuthManager.java

```diff
@@ -128,40 +136,115 @@ boolean resume(AuthorizeResult result) {
         try {
             assertNoError(values.get(KEY_ERROR), values.get(KEY_ERROR_DESCRIPTION));
             assertValidState(parameters.get(KEY_STATE), values.get(KEY_STATE));
-            if (parameters.containsKey(KEY_RESPONSE_TYPE) && parameters.get(KEY_RESPONSE_TYPE).contains(RESPONSE_TYPE_ID_TOKEN)) {
-                assertValidNonce(parameters.get(KEY_NONCE), values.get(KEY_ID_TOKEN));
-            }
+        } catch (AuthenticationException e) {
+            callback.onFailure(e);
+            return true;
+        }
 
-            Log.d(TAG, "Authenticated using web flow");
-            final Date expiresAt = !values.containsKey(KEY_EXPIRES_IN) ? null : new Date(getCurrentTimeInMillis() + Long.valueOf(values.get(KEY_EXPIRES_IN)) * 1000);
-            final Credentials urlCredentials = new Credentials(values.get(KEY_ID_TOKEN), values.get(KEY_ACCESS_TOKEN), values.get(KEY_TOKEN_TYPE), values.get(KEY_REFRESH_TOKEN), expiresAt, values.get(KEY_SCOPE));
-            if (!shouldUsePKCE()) {
-                callback.onSuccess(urlCredentials);
-            } else {
-                //Finish Code Exchange
-                pkce.getToken(values.get(KEY_CODE), new AuthCallback() {
-                    @Override
-                    public void onFailure(@NonNull Dialog dialog) {
-                        callback.onFailure(dialog);
+        final Date expiresAt = !values.containsKey(KEY_EXPIRES_IN) ? null : new Date(getCurrentTimeInMillis() + Long.valueOf(values.get(KEY_EXPIRES_IN)) * 1000);
+        final Credentials frontChannelCredentials = new Credentials(values.get(KEY_ID_TOKEN), values.get(KEY_ACCESS_TOKEN), values.get(KEY_TOKEN_TYPE), values.get(KEY_REFRESH_TOKEN), expiresAt, values.get(KEY_SCOPE));
```

Comment:
Why would `frontChannelCredentials` ever have `values.get(KEY_REFRESH_TOKEN)`? Or token type? Or expires? Or scope? Those are not valid `response_type`s.

Reply:
A refresh token will never be present on the front channel. I've updated that on the `mergeCredentials` method but forgot this line. Will fix 👍 
The rest of the values you mentioned are always present when an access token is returned.

Reply:
I always forget that all those are included in the URL with `response_type=token`